src/app/shared/components/dynamic-field/dynamic-field.component.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    ComponentFactoryResolver,
+    Input,
+    OnInit,
+    ViewContainerRef,
+    Output,
+    EventEmitter,
+    OnDestroy,
+    Component
+  } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { FieldConfig } from '@shared/interfaces/field.interface';
+import { componentMapper } from './form-components';
+
+
+@Component({
+  selector: 'app-dynamic-field',
+  template: '',
+})
+export class DynamicFieldComponent implements OnInit, OnDestroy {
+  @Input() field: FieldConfig;
+  @Input() group: FormGroup;
+
+  @Output() emitHandler: EventEmitter<any> = new EventEmitter();
+
+  destroy$: Subject<boolean> = new Subject<boolean>();
```

Comment:
incorrect name, $ means that this is an Observable return type. Use $$ for Subjects

Reply:
don't agree, can discuss.

Reply:
Discussed